ipalib/x509.py

```diff
@@ -543,3 +544,35 @@ def format_datetime(t):
     if t.tzinfo is None:
         t = t.replace(tzinfo=UTC())
     return unicode(t.strftime("%a %b %d %H:%M:%S %Y %Z"))
+
+
+def match_hostname(cert, hostname):
+    match_cert = {}
+
+    match_cert['subject'] = match_subject = []
+    for rdn in cert.subject.rdns:
+        match_rdn = []
+        for ava in rdn:
+            if ava.oid == cryptography.x509.oid.NameOID.COMMON_NAME:
+                name = 'commonName'
+            else:
+                name = ava.dotted_string
+            match_ava = (name, ava.value)
+            match_rdn.append(match_ava)
+        match_subject.append(match_rdn)
+
+    try:
+        san = cert.extensions.get_extension_for_class(
+                cryptography.x509.SubjectAlternativeName)
+    except cryptography.x509.ExtensionNotFound:
+        pass
+    else:
+        match_cert['subjectAltName'] = match_san = []
+        for value in san.get_values_for_type(cryptography.x509.DNSName):
+            match_dnsname = ('DNS', str(value))
```

Comment:
Why do you use ```str()``` here? It's either unnecessary (if value is already a str) or wrong (if value is bytes or DNSName instance). Since ```get_values_for_type``` returns the value attribute of DNSName instances and DNSName.value is guaranteed to be text, the call is not necessary.